Tracker/Scenes/Trackers Flow/Category settings/CategorySettingsViewController.swift

```diff
@@ -0,0 +1,176 @@
+
+final class CategorySettingsViewController: UIViewController {
+    // MARK: - Properties
+    private let coordinator: SettingsFlowCoordinator
+    private var category: TrackerCategory?
+    private let categories: [TrackerCategory]
+    private let delegate: CategorySettingsViewControllerProtocol
+
+    private var currentCategory: TrackerCategory?
+
+    private lazy var nameTextField: UITextField = {
+        let textField = UITextField()
+        textField.textColor = .Custom.text
+        textField.font = UIFont.systemFont(ofSize: 17)
+        textField.backgroundColor = .Custom.actionBackground
+        textField.placeholder = "Введите название категории"
+        textField.layer.cornerRadius = 16
+        textField.clearButtonMode = .whileEditing
+        textField.leftViewMode = UITextField.ViewMode.always
+        textField.addTarget(self, action: #selector(textFieldDidChange(_:)), for: .editingChanged)
+        textField.delegate = self
+        textField.leftView = UIView(frame:CGRect(x:0, y:0, width:10, height:textField.frame.height))
+        textField.translatesAutoresizingMaskIntoConstraints = false
+        return textField
+    }()
+
+    private let warningLabel: UILabel = {
+        let label = UILabel()
+        label.text = ""
+        label.font = UIFont.systemFont(ofSize: 17)
+        label.textColor = .Custom.red
+        label.textAlignment = .center
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    private let saveButton = CustomButton(title: "Готово")
+
+    // MARK: - Initialiser
+    init(coordinator: SettingsFlowCoordinator, edit category: TrackerCategory?, in categories: [TrackerCategory], delegate: CategorySettingsViewControllerProtocol) {
+        self.coordinator = coordinator
+        self.category = category
+        self.categories = categories
+        self.delegate = delegate
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - LifeCycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+        navigationItem.hidesBackButton = true
+        taps()
+        layout()
+        hideKeyboardWhenTappedAround()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        setupView()
+    }
+
+    // MARK: - Methods
+    private func taps() {
+        saveButton.tapAction = { [weak self] in
+            guard let self = self else { return }
+
+            let newTitle = self.nameTextField.text ?? ""
+            let newCategories = self.updateCategories(with: newTitle)
+            self.delegate.categorySettingsDidUpdated(categories: newCategories)
+            self.coordinator.pop()
+        }
+    }
+
+    private func updateCategories(with title: String) -> [TrackerCategory] {
+        var categoriesUpdated = [TrackerCategory]()
```

Comment:
Можно вот так. Разницы нет, вкусовщина. Но как то принято отделять тип и значение
```suggestion
        var categoriesUpdated: [TrackerCategory] = []
```

Reply:
Если на работе попросят переучиваться то сделаю, пока так уже привык, надеюсь это некритично. Мне кажется логичным не указывать тип при начальном значении. Поэтому когда появляется `=` тип через `:` указывать тип уже не хочется.

Reply:
Не критично, хорошо если на работе есть гайд что так принято :) переучиваться не надо